qiskit_ibm/runtime/runtime_job.py

```diff
@@ -188,6 +188,28 @@ def error_message(self) -> Optional[str]:
         self._set_status_and_error_message()
         return self._error_message
 
+    def wait_for_closed_stream(self, timeout: Optional[float] = None, wait: float = 5) -> None:
+        """If the websocket server stream endpoint is open, wait for it to close.
+
+        Args:
+            timeout: Seconds to wait for the job. If ``None``, wait indefinitely.
+            wait: Seconds between queries.
+
+        Raises:
+            JobTimeoutError: If the job does not reach a final state before the
+                specified timeout.
+        """
+        self._ws_client_future = self._executor.submit(self._start_websocket_client)
```

Comment:
If the user passes callback to IBMRuntimeService.run method then we are already streaming, so we shouldn't start the websocket client here always, we should start it only when we are not streaming already.